pkgs/development/libraries/libcec/default.nix

```diff
@@ -1,33 +1,64 @@
-{ lib, stdenv, fetchurl, cmake, pkg-config, udev, libcec_platform, libraspberrypi ? null }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, udev
+, libcec_platform
+, libraspberrypi ? null
+}:
+let
+  generic = { version, sha256 }: stdenv.mkDerivation rec {
+    pname = "libcec";
+    inherit version;
 
-let version = "4.0.7"; in
+    src = fetchFromGitHub {
+      owner = "Pulse-Eight";
+      repo = pname;
+      rev = "${pname}-${version}";
+      inherit sha256;
+    };
 
-stdenv.mkDerivation {
-  pname = "libcec";
-  inherit version;
+    nativeBuildInputs = [
+      pkg-config
+      cmake
+    ];
 
-  src = fetchurl {
-    url = "https://github.com/Pulse-Eight/libcec/archive/libcec-${version}.tar.gz";
-    sha256 = "0nii8qh3qrn92g8x3canj4glb2bjn6gc1p3f6hfp59ckd4vjrndw";
-  };
+    buildInputs = [
+      udev
+      libcec_platform
+    ] ++ lib.optional (libraspberrypi != null) libraspberrypi;
+
+    cmakeFlags = [ "-DBUILD_SHARED_LIBS=1" ];
 
-  nativeBuildInputs = [ pkg-config cmake ];
-  buildInputs = [ udev libcec_platform ] ++
-    lib.optional (libraspberrypi != null) libraspberrypi;
+    # Fix dlopen path
+    postPatch = ''
+      substituteInPlace include/cecloader.h --replace "libcec.so" "$out/lib/libcec.so"
+    '';
 
-  cmakeFlags = [ "-DBUILD_SHARED_LIBS=1" ];
+    meta = with lib; {
+      description = "USB CEC Adapter communication Library";
+      longDescription = ''
+        Allows you (with the right hardware) to control your device with
+        your TV remote control using existing HDMI cabling.
+      '';
+      homepage = "http://libcec.pulse-eight.com";
+      repositories.git = "https://github.com/Pulse-Eight/libcec.git";
+      license = licenses.gpl2Plus;
+      platforms = platforms.linux;
+      maintainers = with maintainers; [ titanous fab ];
+    };
+  };
 
-  # Fix dlopen path
-  patchPhase = ''
-    substituteInPlace include/cecloader.h --replace "libcec.so" "$out/lib/libcec.so"
-  '';
+in
+rec {
+  libcec_4 = generic {
```

Comment:
Generics are usually split of into another file.

Reply:
I guess then that `libwebsockets` is out-dated.